src/geo.c

```diff
@@ -667,8 +673,8 @@ void georadiusGeneric(client *c, int srcKeyIndex, int flags) {
     }
 
     long result_length = ga->used;
-    long returned_items = (count == 0 || result_length < count) ?
-                          result_length : count;
+    long returned_items = (count == 0 || result_length < llabs(count)) ?
+                          result_length : llabs(count);
```

Comment:
i rather normalize both count and limit to be positive or 0.
have two variables each with it's own meaning, and keep the negative hack for the argument parsing only.
```suggestion
    long returned_items = (count == 0 || result_length < count) ?
                          result_length : count;
```